File: Iteratia1/ExtensieRazboi/src/Card.java
public class Card {

    private int number, color;

    public Card(int number, int color) {
        if(number < 2 || number > 14) {
            System.out.println("Invalid number");
        }
        else this.number = number;

        if(color < 1 || color > 4){
            System.out.println("Invalid color");
        }
        else this.color = color;
    }

    public void setNumber(int number) {
        if(number < 2 || number > 14) {
            System.out.println("Invalid number");
        }
        else this.number = number;
    }

    public void setColor(int color) {
        if(color < 1 || color > 4){
            System.out.println("Invalid color");
        }
        else this.color = color;
    }

    public int getNumber() {
        return number;
    }

    public int getColor() {
        return color;
    }

    @Override
    public String toString() {
        return "Card{" +
                "number=" + number +
                ", color=" + color +
                '}';
    }


}

File: ProjectIP/ProjectIP/Card.cs
using System;
using System.Collections.Generic;
using System.Text;
public enum CardColor//enumul se apeleaza cu CardColor.una din cele 4 valori
{ club, diamond, heart, spade }
namespace ProjectIP
{
    class Card
    {
        private int number;
        private CardColor color;

        //constructor pt carte
        public Card(int number, CardColor color)
        {
            if (number >= 2 && number <= 14)
            {
                this.number = number;
                this.color = color;
            }
            else Console.WriteLine("Invalid card");

        }
        public int getNumber()
        {
            return this.number;
        }
        public void setNumber(int number)
        {
            this.number = number;
        }
        public CardColor getColor()
        {
            return this.color;
        }
        public void setColor(CardColor color)
        {
            this.color = color;
        }
        public override string ToString()
        {
            return "{" + number + "," + color + "}";
        }

    }
}
